### PR TITLE
Fix save errors

### LINKED ISSUE #2: Cannot add new content types with searchable parts to existing Search page
When editing the existing search page, it doesn't pull down the new content types that has been created after the search page got created.

### LINKED ISSUE #3: Creating a search page getting error "The Query field is required"
This is causing when creating a search page with "Grouped - Results are grouped by content type".
This is because the field query field on "List - Results are displayed together in a list" is hidden but still validation happen on that.

### PR DESCRIPTION
**Pull request by mobinzk**
- Fix creating a search page getting error "The Query field is required" (closes #3)
- Fix cannot add new content types with searchable parts to existing Search page (closes #2)

### CODE REVIEW

File: Drivers/SiteSearchDisplay.cs [hunk 1]
```diff
@@ -108,6 +113,11 @@ public async override Task<IDisplayResult> UpdateAsync(SiteSearch part, IUpdateM
                 part.SubmitButtonLabel = model.SubmitButtonLabel;
             }
 
+            if (string.IsNullOrEmpty(part.Query) && part.DisplayType == SiteSearchDisplayType.List)
+            {
+                updater.ModelState.AddModelError(Prefix, nameof(part.Query), T["The Query field is required."]);
```
Comment (peterkeating) [inline Drivers/SiteSearchDisplay.cs]: Can we remove, "The" so it's "Query field is required."

File: Drivers/SiteSearchDisplay.cs [hunk 2]
```diff
@@ -35,17 +38,19 @@ public class SiteSearchPartDisplay : ContentPartDisplayDriver<SiteSearch>
...
         private readonly IQueryManager _queryManager;
+        private readonly IStringLocalizer<SiteSearchPartDisplay> T;
         private readonly YesSql.ISession _session;
 
         #endregion
 
         #region Constructor
 
-        public SiteSearchPartDisplay(IContentDefinitionManager contentDefinitionManager, IHttpContextAccessor httpContextAccessor, IQueryManager queryManager, YesSql.ISession session)
+        public SiteSearchPartDisplay(IContentDefinitionManager contentDefinitionManager, IHttpContextAccessor httpContextAccessor, IQueryManager queryManager, IStringLocalizer<SiteSearchPartDisplay> localizer, YesSql.ISession session)
```
Comment (peterkeating) [inline Drivers/SiteSearchDisplay.cs]: Can the constructor args be alphabetical.

File: Drivers/SiteSearchDisplay.cs [hunk 3]
```diff
@@ -35,17 +38,19 @@ public class SiteSearchPartDisplay : ContentPartDisplayDriver<SiteSearch>
         private readonly IContentDefinitionManager _contentDefinitionManager;
         private readonly IHttpContextAccessor _httpContextAccessor;
         private readonly IQueryManager _queryManager;
+        private readonly IStringLocalizer<SiteSearchPartDisplay> T;
```
Comment (peterkeating) [inline Drivers/SiteSearchDisplay.cs]: Can these be ordered alphabetically, so `T` is last.

File: Drivers/SiteSearchDisplay.cs [hunk 4]
```diff
@@ -35,17 +38,19 @@ public class SiteSearchPartDisplay : ContentPartDisplayDriver<SiteSearch>
...
 
         #region Constructor
 
-        public SiteSearchPartDisplay(IContentDefinitionManager contentDefinitionManager, IHttpContextAccessor httpContextAccessor, IQueryManager queryManager, YesSql.ISession session)
+        public SiteSearchPartDisplay(IContentDefinitionManager contentDefinitionManager, IHttpContextAccessor httpContextAccessor, IQueryManager queryManager, IStringLocalizer<SiteSearchPartDisplay> localizer, YesSql.ISession session)
         {
             _contentDefinitionManager = contentDefinitionManager;
             _httpContextAccessor = httpContextAccessor;
             _queryManager = queryManager;
+            T = localizer;
```
Comment (peterkeating) [inline Drivers/SiteSearchDisplay.cs]: Move this assignment beneath `_session` assignment.

File: Drivers/SiteSearchDisplay.cs [hunk 5]
```diff
@@ -70,7 +70,7 @@ public override async Task<IDisplayResult> DisplayAsync(SiteSearch part, BuildPa
 
         public override async Task<IDisplayResult> EditAsync(SiteSearch part, BuildPartEditorContext context)
         {
-            if (part.ContentTypeSettings == null)
+            if (part.ContentTypeSettings == null || part.ContentTypeSettings.Count() == 0)
```
Comment (peterkeating) [inline Drivers/SiteSearchDisplay.cs]: Instead of `part.ContentTypeSettings.Count() == 0`, use `!part.ContentTypeSettings.Any()`

**Comment by peterkeating** [comment]
@mobinzk Be good if we could get rid of that merge commit.

**Comment by mobinzk** [comment]
@peterkeating I've rebased master into mine and I got that :-/

**Comment by peterkeating** [comment]
> @peterkeating I've rebased master into mine and I got that :-/

That's really odd!

**Comment by mobinzk** [comment]
@peterkeating I've fixed the `merge` you mentioned and other issue with `part.ContentTypeSettings.Any()`

**Pull request merged**